include/seqan3/argument_parser/detail/version_check.hpp

```diff
@@ -26,6 +26,18 @@
 #include <seqan3/std/charconv>
 #include <seqan3/version.hpp>
 
+namespace seqan3
+{
+
+//!\brief Indicated whether application allow automatic update notifications by the seqan3::argument_parser.
+enum class update_notifications
```

Comment:
A publicly visible (non-detail) data structure is defined in a detail header here... I suggest moving it to `include/seqan3/argument_parser/argument_parser.hpp`?

Reply:
Good point, I will move it.

Reply:
Note that I could not move it to argument_parser.hpp because argument_parser.hpp depends on version_check.hpp and version_check.hpp needs the `update_notifications` struct so this would create dependency loop. I put it into auxiliary instead.